lib/assets/javascripts/twine.coffee

```diff
@@ -46,33 +46,47 @@
     this
 
   Twine.bind = (node = rootNode, context = Twine.context(node)) ->
-    bind(context, node, true)
+    bind(context, node, getIndexesForElement(node), true)
 
   Twine.afterBound = (callback) ->
     if currentBindingCallbacks
       currentBindingCallbacks.push(callback)
     else
       callback()
 
-  bind = (context, node, forceSaveContext) ->
+  bind = (context, node, indexes, forceSaveContext) ->
     currentBindingCallbacks = []
     if node.bindingId
       Twine.unbind(node)
 
+    if defineArrayAttr = Twine.getAttribute(node, 'define-array')
+      newIndexes = defineArray(node, context, defineArrayAttr)
+      for key, value of indexes || {} when !newIndexes[key]?
```

Comment:
Can you pull the `indexes || {}` part up to a dedicated variable? This line is really hard to parse. Also, `hasOwnProperty` is probably more what you are after for the `newIndexes[key]` part.